password-hash/src/salt.rs

```diff
@@ -171,13 +183,15 @@ impl<'a> fmt::Debug for Salt<'a> {
 }
 
 /// Owned stack-allocated equivalent of [`Salt`].
+///
+/// Internally encoded as B64 characters
 #[derive(Clone, Eq)]
 pub struct SaltString {
-    /// Byte array containing an ASCiI-encoded string.
-    bytes: [u8; Salt::MAX_LENGTH],
+    /// Byte array containing an ASCiI-encoded string, length always matches the byte length.
+    characters: [u8; Salt::MAX_LENGTH],
 
-    /// Length of the string in ASCII characters (i.e. bytes).
-    length: u8,
+    /// Length of the string in ASCII characters in Bytes.
+    ascii_length: usize,
```

Comment:
Not sure this is an improvement. `ascii_length` is the same as the length in bytes.